File: 백준/Silver/2606. 바이러스/바이러스.cc
#include <bits/stdc++.h>
using namespace std;

int N, M, answer = -1;
bool visited[101];
vector<int> graph[101];

void dfs(int node) {
	visited[node] = true;
	answer++;
	for (int i = 0; i < graph[node].size(); i++) {
		int next = graph[node][i];
		if (!visited[next]) dfs(next);
	}
}


int main() {
	ios_base::sync_with_stdio(0); cin.tie(0); cout.tie(0);
	cin >> N >> M;
	int x, y;
	for (int i = 0; i < M; i++) {
		cin >> x >> y;
		graph[x].push_back(y);
		graph[y].push_back(x);
	}
	
	dfs(1);

	cout << answer;

	return 0;
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys
sys.setrecursionlimit(10**6)

def dfs(graph, visit, node, cnt = -1):
    visit[node] = True
    
    for next in graph[node]:
        if not visit[next]:
            cnt = dfs(graph, visit, next, cnt)
    
    return cnt + 1
def main():
    v, e = int(input()), int(input())
    graph = [[] for _ in range(v + 1)]
    visit = [False] * (v + 1)
    
    for _ in range(e):
        start, end = map(int, input().split())
        graph[start].append(end)
        graph[end].append(start)
    
    answer = dfs(graph, visit, 1)
    print(answer)
if __name__ == "__main__":
    main()